app/src/main/java/com/hedvig/app/feature/settings/Language.kt

```diff
@@ -0,0 +1,124 @@
+package com.hedvig.app.feature.settings
+
+import android.annotation.TargetApi
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Build
+import android.os.LocaleList
+import androidx.preference.PreferenceManager
+import java.util.Locale
+
+enum class Language {
+    SYSTEM_DEFAULT,
+    SV_SE,
+    EN_SE;
+
+    fun apply(context: Context?): Context? {
+        val locale = into()
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            applyV24(context, locale)
+        } else {
+            applyV21(context, locale)
+        }
+    }
+
+    private fun applyV21(context: Context?, locale: LocaleWrapper): Context? {
+        if (locale !is LocaleWrapper.SingleLocale) {
+            throw RuntimeException("Invalid state: API version <= 21 but multiple locales was encountered")
+        }
+        val unwrappedLocale = locale.locale
+        Locale.setDefault(unwrappedLocale)
+        if (context == null) {
+            return null
+        }
+
+        val config = Configuration(context.resources.configuration)
+        config.setLocale(unwrappedLocale)
+        context.resources.updateConfiguration(config, context.resources.displayMetrics)
+        return context.createConfigurationContext(config)
+    }
+
+    @TargetApi(Build.VERSION_CODES.N)
+    private fun applyV24(context: Context?, locale: LocaleWrapper): Context? {
+        when (locale) {
+            is LocaleWrapper.SingleLocale -> {
+                val unwrappedLocale = locale.locale
+                Locale.setDefault(unwrappedLocale)
+                if (context == null) {
+                    return null
+                }
+                val config = Configuration(context.resources.configuration)
+                config.setLocale(unwrappedLocale)
+                return context.createConfigurationContext(config)
+            }
+            is LocaleWrapper.MultipleLocales -> {
+                val locales = locale.locales
+                LocaleList.setDefault(locales)
+                if (context == null) {
+                    return null
+                }
+                val config = Configuration(context.resources.configuration)
+                config.setLocales(locales)
+                return context.createConfigurationContext(config)
```

Comment:
this is basically the same code as line 47-52 maybe you could put in a function, I't also pretty similar to 30-38 so have a look at that as well